File: 2018/day2/part2.exs
lines =
  File.stream!("input.txt")
  |> Stream.map(&String.trim/1)
  |> Stream.map(&String.graphemes/1)

Enum.map(lines, fn id1 ->
  Enum.map(lines, fn id2 ->
    result = Enum.zip(id1, id2) |> Enum.map(fn {x, y} -> x == y end)

    if Enum.count(result, fn x -> x == false end) == 1 do
      idx = Enum.find_index(result, fn x -> x == false end)

      IO.inspect(List.delete_at(id1, idx) |> Enum.join())
      exit(:shutdown)
    end
  end)
end)

File: 2018/day2/part2.rb
ids = File.readlines("input.txt").map(&:chomp).map(&:chars)

ids.each_with_index do |id1, i|
  ids[(i+1)..-1].each do |id2|
    result = id1.zip(id2).map {|l| l.uniq.length == 1}
    if result.count(false) == 1
      puts id1.reduce("", :+)
      abort
    end
  end
end
